### PR TITLE
Main branch edge to edge implementation

### PR DESCRIPTION
**Pull request by the-scrambler**
Merge pending codelab update.

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -15,18 +15,18 @@
...
 plugins {
     id("com.android.application")
-    id("com.google.devtools.ksp") version "1.8.21-1.0.11"
+    id("com.google.devtools.ksp") version "1.9.0-1.0.13"
     id("org.jetbrains.kotlin.android")
 }
 
 android {
     namespace = "com.example.busschedule"
-    compileSdk = 33
```
Comment (android-dev-lxl) [inline app/build.gradle.kts:24]: Since Giraffe template code uses SDK 33, Murat suggested setting target SDK to 33 for now.

Comment (android-dev-lxl) [reply app/build.gradle.kts:24]: This requires lowering other version as well. 
Is there a specific reason to do this?

File: app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt [hunk 2]
```diff
@@ -167,14 +191,24 @@ fun BusScheduleScreen(
         Row(
             modifier = Modifier
                 .fillMaxWidth()
-                .padding(dimensionResource(R.dimen.padding_medium)),
+                .padding(dimensionResource(R.dimen.padding_medium))
```
Comment (johnshea) [inline app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt:194]: Can these two `.padding()` function calls be combined into one or one of them removed?

**Review by johnshea — APPROVED** [review]
LGTM with two small issues.

- Upgrade gradle wrapper to `8.2`
- Address `.padding()` issue noted in file

File: app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt [hunk 3]
```diff
@@ -109,12 +119,18 @@ fun BusScheduleApp(
...
                 val routeSchedule by viewModel.getScheduleFor(stopName).collectAsState(emptyList())
                 RouteScheduleScreen(
                     stopName = stopName,
+                    modifier = Modifier.padding(
+                        start = innerPadding.calculateStartPadding(layoutDirection),
+                        end = innerPadding.calculateEndPadding(layoutDirection),
+                    ),
                     busSchedules = routeSchedule,
+                    contentPadding = innerPadding,
                     onBack = { onBackHandler() }
```
Comment (android-dev-lxl) [inline app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt:134]: this is bit confusing. 
We either use the content padding as a `Modifier.padding` or as `contentPadding` not both.

File: app/src/main/java/com/example/busschedule/ui/theme/Theme.kt [hunk 4]
```diff
@@ -125,3 +125,19 @@ fun BusScheduleTheme(
...
+
+private fun setUpEdgeToEdge(view: View, darkTheme: Boolean) {
+    val window = (view.context as Activity).window
+    WindowCompat.setDecorFitsSystemWindows(window, false)
+    window.statusBarColor = Color.Transparent.toArgb()
+    val navigationBarColor = when {
+        Build.VERSION.SDK_INT >= 29 -> Color.Transparent.toArgb()
+        Build.VERSION.SDK_INT >= 26 -> Color(0xFF, 0xFF, 0xFF, 0x63).toArgb()
+        // Min sdk version for this app is 24, this block is for SDK versions 24 and 25
+        else -> Color(0x00,0x00, 0x00, 0x50).toArgb()
```
Comment (android-dev-lxl) [inline app/src/main/java/com/example/busschedule/ui/theme/Theme.kt:137]: Nit: Add space before second parameter 
else -> Color(0x00, 0x00, 0x00, 0x50).toArgb()

File: build.gradle.kts [hunk 5]
```diff
@@ -16,12 +16,12 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
     extra.apply {
-        set("nav_version", "2.5.3")
-        set("room_version", "2.5.1")
+        set("nav_version", "2.7.0")
```
Comment (android-dev-lxl) [inline build.gradle.kts:19]: Are we not supposed tho rollback to older version?

**Review by android-dev-lxl — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt [hunk 6]
```diff
@@ -191,9 +220,10 @@ fun BusScheduleScreen(
 fun BusScheduleDetails(
     busSchedules: List<BusSchedule>,
     modifier: Modifier = Modifier,
```
Comment (android-dev-lxl) [inline app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt:222]: Similar to above.
Pass this from the parent and please use it.

File: app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt [hunk 7]
```diff
@@ -90,12 +95,13 @@ fun BusScheduleApp(
     ) { innerPadding ->
         NavHost(
             navController = navController,
-            modifier = Modifier.padding(innerPadding),
             startDestination = BusScheduleScreens.FullSchedule.name
         ) {
             composable(BusScheduleScreens.FullSchedule.name) {
+                val layoutDirection = LocalLayoutDirection.current
```
[the remark this answers is not in the document]
Comment (android-dev-lxl) [reply app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt:101]: also the import

File: app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt [hunk 8]
```diff
@@ -41,6 +44,7 @@ import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalLayoutDirection
```
Comment (android-dev-lxl) [inline app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt:47]: I don't think this is needed.

**Review by android-dev-lxl — DISMISSED** [review]

**Review by johnshea — CHANGES_REQUESTED** [review]
Changes were made after I approved this PR that are negatively affecting the UI so I am now marking as "Request changes".

I reviewed SHA 65e9b95c7dec59cc49829b70b273bc26b6bd42a8
Screenshot 2023-08-29 at 1 59 18 PM

Current SHA 0455a819

Screenshot 2023-08-29 at 1 57 06 PM

Portrait mode (I reviewed left most image, `main` is on right, `main-e2e` is in the middle)
The spacing at the top is now off in the latest commit.
Screenshot 2023-08-29 at 2 04 57 PM

File: build.gradle.kts [hunk 9]
```diff
@@ -17,11 +17,11 @@
...
-        set("room_version", "2.5.1")
+        set("room_version", "2.5.2")
     }
 }
 plugins {
-    id("com.android.application") version "8.0.2" apply false
-    id("com.android.library") version "8.0.2" apply false
-    id("org.jetbrains.kotlin.android") version "1.8.21" apply false
+    id("com.android.application") version "8.1.0" apply false
+    id("com.android.library") version "8.1.0" apply false
```
Comment (johnshea) [inline build.gradle.kts:25]: Update to latest version. Currently `8.1.1`.

File: build.gradle.kts [hunk 10]
```diff
@@ -17,11 +17,11 @@
...
         set("nav_version", "2.5.3")
-        set("room_version", "2.5.1")
+        set("room_version", "2.5.2")
     }
 }
 plugins {
-    id("com.android.application") version "8.0.2" apply false
-    id("com.android.library") version "8.0.2" apply false
-    id("org.jetbrains.kotlin.android") version "1.8.21" apply false
+    id("com.android.application") version "8.1.0" apply false
```
Comment (johnshea) [inline build.gradle.kts:24]: Update to latest version. Currently `8.1.1`.

File: build.gradle.kts [hunk 11]
```diff
@@ -17,11 +17,11 @@
...
+    id("com.android.library") version "8.1.0" apply false
+    id("org.jetbrains.kotlin.android") version "1.9.0" apply false
```
Comment (johnshea) [inline build.gradle.kts:26]: Update to latest version. Currently `1.9.10`.

File: app/build.gradle.kts [hunk 12]
```diff
@@ -51,13 +51,12 @@ android {
...
     kotlinOptions {
         jvmTarget = "17"
-        freeCompilerArgs += "-opt-in=androidx.compose.material3.ExperimentalMaterial3Api"
     }
     buildFeatures {
         compose = true
     }
     composeOptions {
-        kotlinCompilerExtensionVersion = "1.4.7"
+        kotlinCompilerExtensionVersion = "1.5.1"
```
Comment (johnshea) [inline app/build.gradle.kts:59]: Update to latest. Currently `1.5.2`.

File: app/src/main/java/com/example/busschedule/ui/theme/Theme.kt [hunk 13]
```diff
@@ -125,3 +125,19 @@ fun BusScheduleTheme(
         content = content
     )
 }
+
+private fun setUpEdgeToEdge(view: View, darkTheme: Boolean) {
```
Comment (johnshea) [inline app/src/main/java/com/example/busschedule/ui/theme/Theme.kt:129]: Add the comment block.

File: app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt [hunk 14]
```diff
@@ -162,19 +172,33 @@ fun BusScheduleScreen(
...
+            start = contentPadding.calculateStartPadding(layoutDirection),
+            end = contentPadding.calculateEndPadding(layoutDirection),
+        )
+    ) {
         Row(
             modifier = Modifier
                 .fillMaxWidth()
-                .padding(dimensionResource(R.dimen.padding_medium)),
+                .padding(
+                    top = contentPadding.calculateTopPadding(),
```
Comment (johnshea) [inline app/src/main/java/com/example/busschedule/ui/BusScheduleScreens.kt:186]: nit: Rearrange arguments so they match the function signature.

Screenshot 2023-08-31 at 3 51 21 PM

**Review by johnshea — DISMISSED** [review]
In general LGTM so approving.

Please upgrade versions to the latest and consider fixing the nit.

**Pull request closed without merging**